### ISSUE TITLE
dotnet build for ASP.NET Core app on macOS fails due to missing Microsoft.NETCore.App package for RC1

### ISSUE THREAD

**Issue by martincostello** [dotnet/cli#9215]
## Steps to reproduce

Build justeat/httpclient-interception#23 at [e53fd38](https://github.com/justeat/httpclient-interception/pull/23/commits/e53fd389ce8161f4bbe802d8fe28eb59fe4899e5) for macOS in Travis CI.

## Expected  behavior

Build succeeds.

## Actual behavior

Build fails [with error](https://travis-ci.org/justeat/httpclient-interception/jobs/376103297#L226):

```sh
/Users/travis/build/justeat/httpclient-interception/samples/SampleApp/SampleApp.csproj : error NU1102: Unable to find package Microsoft.NETCore.App with version (>= 2.1.0-rc1-26428-03) [/Users/travis/build/justeat/httpclient-interception/samples/SampleApp.Tests/SampleApp.Tests.csproj]
/Users/travis/build/justeat/httpclient-interception/samples/SampleApp/SampleApp.csproj : error NU1102:   - Found 32 version(s) in nuget.org [ Nearest version: 2.1.0-rc1 ] [/Users/travis/build/justeat/httpclient-interception/samples/SampleApp.Tests/SampleApp.Tests.csproj]
/Users/travis/build/justeat/httpclient-interception/samples/SampleApp.Tests/SampleApp.Tests.csproj : error NU1102: Unable to find package Microsoft.NETCore.App with version (>= 2.1.0-rc1-26428-03)
/Users/travis/build/justeat/httpclient-interception/samples/SampleApp.Tests/SampleApp.Tests.csproj : error NU1102:   - Found 32 version(s) in nuget.org [ Nearest version: 2.1.0-rc1 ]
```

The same commit builds successfully in Travis CI for Linux and in AppVeyor for Windows.

**Comment by livarcocc** [comment]
Hum, that is the right rc1 version. Do you have the RuntimeFrameworkVersion explicitly set in your project? I just tried that build and I see (at least for a console app) that it references 2.1.0-rc1 of Microsoft.NETCore.App and not the version you shared above.

**Comment by martincostello** [comment]
I’m targeting `netcoreapp2.1` in the sample project’s .csproj file. I assumed that the rc1 NuGet packages would have synced up ok with the RC1 SDK. The approach is the same as I did for Previews 1 & 2.

**Comment by livarcocc** [comment]
Can you try cleaning your project and re-restoring?

**Comment by martincostello** [comment]
Given that Travis CI and AppVeyor are _supposed_ to provide clean images for every build, that wouldn't explain the issues.

**Comment by livarcocc** [comment]
Can you share the project.assets.json or a binary log for the failing build? So that we can try to identify where that version is coming from? It shouold be referencing 2.1.0-rc1.

**Comment by martincostello** [comment]
Sure - once I'm done at Build today I'll look at turning on binary logging in the PR with the issue.

**Comment by martincostello** [comment]
Man, Travis CI don't make it easy to get files out of builds...

**Comment by parekhkb** [comment]
@martincostello check out #9231, I was getting the exact same error  `error NU1102: Unable to find package Microsoft.NETCore.App with version (>= 2.1.0-rc1-26428-03)`

**Comment by martincostello** [comment]
Yep, that was it. I’ve just re-run the macOS Travis CI build and now it passes like the Windows and Linux ones.
